data/Store.js

```diff
@@ -217,8 +217,13 @@ export class Store {
         return this._all.rootList;
     }
 
-    /** Filter function to be applied. */
     get filter() {return this._filter}
+    /**
+     * Filter function to be applied.
+     * @param {Function} fn - function taking a Record, and returning a boolean.
+     * @param {boolean} [fn.includeParents] - true to include all ancestors of a passing record (default true)
```

Comment:
I think this is neat and encapsulates the props nicely in the function, but seems like this could result in some clunky code in applications trying to use this. I often pass an arrow function or just set the filter to a class member, so if I understand this correctly I would need to do something like this:
```
this.myFilterFn.includeChildren = true;
myStore.setFilter(this.myFilterFn);
```
or 
```
const filterFn = (rec) => rec.passesMyCondition;
filterFn.includeChildren = true;
myStore.setFilter(filterFn);
```
as opposed to:
```
myStore.setFilter(this.filterFn, {includeChildren: true});
```
or
```
myStore.setFilterFn((rec) => rec.passesMyCondition, {includeChildren: true})
```

Is there some magical syntax sugar that would make this cleaner including the props on the function object?

Reply:
John -- thanks for this comment.  It is SO important to actually write the application code that your new API is intended to support.  I totally failed to do that.  I agree the property on the filter is way too clumsy looking.  I'll change to the option you mention